src/controllers/ExtensionController.ts

```diff
@@ -296,6 +297,21 @@ export class ExtensionController implements Disposable {
     );
   };
 
+  /**
+   * Listen to events that will enable / disable server polling.
+   */
+  initializePolling = (): void => {
+    assertDefined(this._serverTreeView, 'serverManager');
+
+    vscode.window.onDidChangeWindowState(
+      this.onUpdateServerPolling,
+      undefined,
+      this._context.subscriptions
+    );
+
+    this._serverTreeView.onDidChangeVisibility(this.onUpdateServerPolling);
```

Comment:
Do we need to call `onUpdateServerPolling` here so that it starts polling if we're already in the foreground/serverTreeView is ready?

Reply:
It is currently safe since `ServerManager.loadServerConfig` is calls it it the ServerManager constructor. That said, since that is an implementation detail that the extension controller is unaware of, it makes sense to call it explicitly here as you are suggesting.

Reply:
I've updated.